sno/pk_generation.py

```diff
@@ -406,3 +452,45 @@ def feature_blobs(self):
             pk = self.decode_path_to_1pk(blob.name)
             if self.pk_filter(pk):
                 yield blob
+
+
+class FeaturePlusHashes:
+    _MORE_THAN_ONE_MATCH = object()
```

Comment:
worth adding [`__slots__`](https://docs.python.org/3.9/reference/datamodel.html#invoking-descriptors) here since we're potentially creating a lot of them?